reps/2023-07-08-air-surface-syntax.md

```diff
@@ -0,0 +1,114 @@
+
+### Should this change be within `ray` or outside?
+
+main `ray` project. Changes are made to Ray Train, Tune and AIR.
+
+## Stewardship
+
+### Required Reviewers
+The proposal will be open to the public, but please suggest a few experienced Ray contributors in this technical domain whose comments will help this proposal. Ideally, the list should include Ray committers.
+
+@matthewdeng, @krfricke
+
+### Shepherd of the Proposal (should be a senior committer)
+To make the review process more productive, the owner of each proposal should identify a **shepherd** (should be a senior Ray committer). The shepherd is responsible for working with the owner and making sure the proposal is in good shape (with necessary information) before marking it as ready for broader review.
+
+@ericl
+
+## Details of the API changes
+
+Concretely, we replace the Ray AIR session with a training context to
+1. avoid the user confusion of what a `session` is (and not having to explain in the documentation) and
+2. bring the API in line with other Ray APIs like `get_runtime_context` as well as Ray Data's `DataContext`.
+
+The API changes are
+```
+from ray import air, train
+
+# Ray Train methods and classes:
+
+air.session.report               -> train.report
+air.session.get_dataset_shard    -> train.get_dataset_shard
+air.Checkpoint                   -> train.Checkpoint
+air.Result                       -> train.Result
+
+# Ray Train configurations:
+
+air.config.CheckpointConfig      -> train.CheckpointConfig
+air.config.FailureConfig         -> train.FailureConfig
+air.config.RunConfig             -> train.RunConfig
+air.config.ScalingConfig         -> train.ScalingConfig
+
+# Ray TrainContext methods:
+
+air.session.get_checkpoint       -> train.get_context().get_checkpoint
+air.session.get_experiment_name  -> train.get_context().get_experiment_name
+air.session.get_trial_name       -> train.get_context().get_trial_name
+air.session.get_trial_id         -> train.get_context().get_trial_id
+air.session.get_trial_resources  -> train.get_context().get_trial_resources
+air.session.get_trial_dir        -> train.get_context().get_trial_dir
+air.session.get_world_size       -> train.get_context().get_world_size
+air.session.get_world_rank       -> train.get_context().get_world_rank
+air.session.get_local_rank       -> train.get_context().get_local_rank
+air.session.get_local_world_size -> train.get_context().get_local_world_size
+air.session.get_node_rank        -> train.get_context().get_node_rank
+
+del air
+```
+
+These changes are ready to try out with https://github.com/ray-project/ray/pull/36706 and we encourage user feedback on the changes.
+
+## Open Questions
+
+We are likely going to remove `PredictorWrapper` and `PredictorDeployment` and migrate the examples to use Ray Serve deployments
+direcly, and we are also likely going to move `air.integrations` to `train.integrations` and tentatively the predictors to `ray.train`.
```

Comment:
There's a few more:

- `air.callbacks` --> also train?
- `air.examples` for cross-library examples --> docs? (related question: how do we restructure the docs? I guess out of scope for this REP...)
- `air.execution` --> tentatively train?
- `air.util` - looks like mostly data?

Reply:
Would it make sense to capture all these in the REP to make it more comprehensive, i.e. to describe what it takes to fully "Disband the `ray.air` namespace"?